CONTRIBUTING.md

```diff
@@ -164,189 +331,275 @@ pgbackrest/test/test.pl --vm=u18 --dev --module=mock --test=archive --run=2
+Unit test files are organized in the `test/src/module` directory with the same directory structure as the source code being tested. For example, if new code is added to src/**command/expire**.c then test/src/module/**command/expire**Test.c will need to be updated.
 
-- **Section 2:** Find the second section with the `Command-line only options` comment. This is where the rules for the option are defined.
+Assuming that a test file already exists, new unit tests will either go in a new `testBegin()` section or be added to an existing section. Each such section is a test run. The comment string passed to the `testBegin()` should reflect the function(s) being tested in the test run. Tests within a run should use `TEST_TITLE()` with a comment string describing the test.
+```
+// *****************************************************************************************************************************
+if (testBegin("expireBackup()"))
+{
+    //--------------------------------------------------------------------------------------------------------------------------
+    TEST_TITLE("manifest file removal");
+```
 
-The steps for how to update these sections are detailed below.
+#### Setting up the command to be run
 
-**Section 1**
+The [harnessConfig.h](https://github.com/pgbackrest/pgbackrest/blob/master/test/src/common/harnessConfig.h) describes a list of functions that should be used when configuration options are required for a command being tested. Options are set in a `StringList` which must be defined and passed to the function `harnessCfgLoad()` with the command. For example, the following will set up a test to run `pgbackrest --repo-path=test/test-0/repo info` command:
+```
+StringList *argList = strLstNew();                              // Create an empty string list
+hrnCfgArgRawZ(argList, cfgOptRepoPath, TEST_PATH_REPO);         // Add the --repo-path option
+harnessCfgLoad(cfgCmdInfo, argList);                            // Load the command and option list into the test harness
 
-Copy the two lines ("use constant"/"push") of an existing option and paste them where the option would be in alphabetical order and rename it to the same name as the new option name. For example CFGOPT_DRY_RUN, defined as "dry-run".
+// Run the test
+TEST_RESULT_STR_Z(
+    infoRender(), "No stanzas exist in the repository.\n", "text output - no stanzas");
+```
 
-**Section 2**
+#### Storing a file
 
-To better explain this section, `CFGOPT_ONLINE` will be used as an example:
+Sometimes it is desirable to store or manipulate files before or during a test and then confirm the contents. The [harnessStorage.h](https://github.com/pgbackrest/pgbackrest/blob/master/test/src/common/harnessStorage.h) file contains macros (e.g. `HRN_STORAGE_PUT` and `TEST_STORAGE_GET`) for doing this. In addition, `HRN_INFO_PUT` is convenient for writing out info files (archive.info, backup.info, backup.manifest) since it will automatically add header and checksum information.
```

Comment:
Does this mean that `HRN_STORAGE_PUT` should be used everywhere instead of `TEST_RESULT_VOID(storagePutP(...` or is there some caveats where both should/could be used?

Reply:
We're trying to encourage everyone to use the newer macros - the old ways still work, but we'd rather not detail the history of the progression to the macros - rather just outline the way it should be done. Maybe we can work on a project to update the tests to be more consistent throughout...

Reply:
Seen the working example in `repo-retention-history` PR. It would perhaps be interesting to update the tests to be consistent but stating that we're now only adding/using new macros is fine :-)